Frontend/src/app/market/market.component.ts

```diff
@@ -58,38 +62,61 @@ export class MarketComponent implements OnInit{
   submitForm() {
       // Your implementation for submitForm
     }
-    initForm() {
-      this.filterForm = this.formBuilder.group({
-        country: [''],
-        state: [''],
-        city: ['']
-      });
-    }
   
-    loadCountries() {
-      //this.dataService.getCountries().subscribe((data: any[]) => {
-        //this.countries = data;
-      //});
-    }
 
-    onCountryChange() {
-      /*const selectedCountryId = this.filterForm.get('country').value;
-      // Load states based on the selected country
-      this.dataService.getStates(selectedCountryId).subscribe((data: any[]) => {
-        this.states = data;
-        this.filterForm.get('state').setValue(''); // Reset state selection
-        this.filterForm.get('city').setValue(''); // Reset city selection
-      });*/
-    }
   
-    onStateChange() {
-      /*const selectedStateId = this.filterForm.get('state').value;
-      // Load cities based on the selected state
-      this.dataService.getCities(selectedStateId).subscribe((data: any[]) => {
-        this.cities = data;
-        this.filterForm.get('city').setValue(''); // Reset city selection
-      });*/
-    }
+    onSubmit() {
+      // Assign form values to variables
+      const country = this.form.get('country')?.value;
+      const state = this.form.get('state')?.value;
+      const city = this.form.get('city')?.value;
+  
+      // Store the data in the object
+      this.submittedData = {
+        country: country,
+        state: state,
+        city: city
+      };
+console.log(this.submittedData )
+/*await this.defaultService.httpPostCall(,).subscribe(
```

Comment:
Remove commented code if not being used

Reply:
They required for retriving data from backend, So it can be easy for you to use them. I am getting error because, I don't have that API to write that bits of code.

Reply:
Okay